build.gradle

```diff
@@ -141,92 +142,79 @@ task testExternalApis(type: Test) {
-if (JavaVersion.current().isJava8Compatible()) {
-    tasks.withType(Javadoc) {
-        options.addStringOption('Xdoclint:none', '-quiet')
-    }
+tasks.withType(Javadoc) {
+    options.addStringOption('Xdoclint:none', '-quiet')
 }
 
 task javadocJar(type: Jar, dependsOn: javadoc) {
-    classifier = 'javadoc'
+    archiveClassifier.set('javadoc')
     from 'build/docs/javadoc'
 }
 
 task sourcesJar(type: Jar) {
     from sourceSets.main.allSource
-    classifier = 'sources'
-}
-
-/**
- * This specifies what artifacts will be built and uploaded when performing a maven upload.
- */
-artifacts {
-    archives jar
-    archives javadocJar
-    archives sourcesJar
+    archiveClassifier.set('sources')
 }
 
 /**
  * Sign non-snapshot releases with our secret key.  This should never need to be invoked directly.
  */
 signing {
-    required { isRelease && gradle.taskGraph.hasTask("uploadArchives") }
+    required { isRelease && gradle.taskGraph.hasTask("publishHtsjdkPublicationToMavenRepository") }
     sign configurations.archives
 }
 
-/**
- * Upload a release to sonatype.  You must be an authorized uploader and have your sonatype
- * username and password information in your gradle properties file.  See the readme for more info.
- *
- * For releasing to your local maven repo, use gradle install
- */
-uploadArchives {
-    repositories {
-        mavenDeployer {
-            beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
+publishing {
+    publications {
+        htsjdk(MavenPublication) {
+            from components.java
+            artifact javadocJar
+            artifact sourcesJar
 
-            repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
-                authentication(userName: project.findProperty("sonatypeUsername"), password: project.findProperty("sonatypePassword"))
-            }
-
-            snapshotRepository(url: "https://broadinstitute.jfrog.io/broadinstitute/libs-snapshot-local/") {
-                authentication(userName: System.env.ARTIFACTORY_USERNAME, password: System.env.ARTIFACTORY_PASSWORD)
-            }
-
-            pom.project {
-                name 'HTSJDK'
-                packaging 'jar'
-                description 'A Java API for high-throughput sequencing data (HTS) formats'
-                url 'http://samtools.github.io/htsjdk/'
+            pom {
+                name = 'HTSJDK'
+                packaging = 'jar'
+                description = 'A Java API for high-throughput sequencing data (HTS) formats'
+                url = 'http://samtools.github.io/htsjdk/'
 
                 developers {
                     developer {
-                        id 'picard'
-                        name 'Picard Team'
-                        url 'http://broadinstitute.github.io/picard'
+                        id = 'picard'
+                        name = 'Picard Team'
+                        url = 'http://broadinstitute.github.io/picard'
```

Comment:
Unrelated and tangential, but this seems like ancient history. And if you look there, you see nothing about htsjdk.

Reply:
Yeah, I noticed that too but decided to leave it for later...  I'll open a follow up to change it.  Not sure to what though...